docs/DeveloperGuide.md

```diff
@@ -1,34 +1,422 @@
+
+![image-20200326014336120](images/Add_Module_Command_Class_Diagram.png)
+
+<span style="color: green"><small><i>Figure <b>Add Module Command Class Diagram</b></i></small></span>
+
+#### Example Usage
+
+The addition of modules will be illustrated as follows.
+
+James is a user and wants to add the module with the module code of "CS3235". Assume that he has the current modules:
+
+```
++--------------------------------------------------------------------------------------------------+
+ NO |  MODULE CODE   |                                 MODULE TITLE
++--------------------------------------------------------------------------------------------------+
+ 1  |     CS1231     |                             Discrete Structures
+ 2  |     CS2100     |                            Computer Organisation
+ 3  |     CS2113     |              Software Engineering & Object-Oriented Programming
++--------------------------------------------------------------------------------------------------+
+Total modules: 3
++--------------------------------------------------------------------------------------------------+
+```
+
+1. James will simply enter the command `addm cs3235`
+
+   After the input is parsed as an **add module task** and executed, the `AddModuleCommand#execute()` will call `ModuleManager#add()` to add the module cs3235. In the `ModuleManager#add()` method, it will call `ModuleManager#contains()` to check if the module cs3235 exists in the `ArrayList` named `moduleList` , then it will check if the module code "cs3235" is a key in the `HashMap` named `modulesMap`, after all, it will instantiate an `Module` object with the module code "CS3235" and respective title "Computer Security", then add the object into the `moduleList`.
+
+2. James receive the following feedback:
+
+   ```
+   root :
+   addm cs3235
+   SUCCESS!! Module CS3235 Computer Security has been added.
+   
+   root :
+   lsm
+   Here are what you are looking for...
+   
+   +--------------------------------------------------------------------------------------------------+
+    NO |  MODULE CODE   |                                 MODULE TITLE
+   +--------------------------------------------------------------------------------------------------+
+    1  |     CS1231     |                             Discrete Structures
+    2  |     CS2100     |                            Computer Organisation
+    3  |     CS2113     |              Software Engineering & Object-Oriented Programming
+    4  |     CS3235     |                              Computer Security
+   +--------------------------------------------------------------------------------------------------+
+   Total modules: 4
+   +--------------------------------------------------------------------------------------------------+
+   ```
+
+
+
+Below is a *sequence diagram* to illustrate the above example scenario.  
+
+```
+	// To do Sequence diagram here
+```
+
+
+
+#### 1.2. Add Category and Task Features
```

Comment:
Maybe can add some more visuals to this section? It was a bit hard for me to follow just reading along